chia/server/start_data_layer.py

```diff
@@ -22,7 +22,11 @@
 
 # TODO: Review need for config and if retained then hint it properly.
 def service_kwargs_for_data_layer(
-    root_path: pathlib.Path, config: Dict[str, Any], wallet_rpc_port: Optional[uint16] = None
+    root_path: pathlib.Path,
+    config: Dict[str, Any],
+    port: Optional[int] = None,
+    rpc_port: Optional[int] = None,
+    wallet_rpc_port: Optional[int] = None,
```

Comment:
the fact that you pass both the `config` and a series of optional ports seem to suggest (to me at least) that the picking of ports should happen one layer up, and set them in the `config` object. I suppose there's a good reason to do it at this level.

This is also "production code" which ideally would have minimal impact by test logic.